core/src/ast_builder/select/mod.rs

```diff
@@ -0,0 +1,62 @@
+mod group_by;
+mod having;
+mod limit;
+mod limit_offset;
+mod offset;
+mod offset_limit;
+mod project;
+mod root;
+
+pub use {
+    group_by::GroupByNode, having::HavingNode, limit::LimitNode, limit_offset::LimitOffsetNode,
+    offset::OffsetNode, offset_limit::OffsetLimitNode, project::ProjectNode, root::SelectNode,
+};
+
+use crate::{
+    ast::{Expr, Query, Select, SelectItem, SetExpr, Statement, TableWithJoins},
+    result::Result,
+};
+
+struct NodeData {
+    pub projection: Vec<SelectItem>,
+    pub from: TableWithJoins,
+    /// WHERE
+    pub selection: Option<Expr>,
+    pub group_by: Vec<Expr>,
+    pub having: Option<Expr>,
+    pub limit: Option<Expr>,
+    pub offset: Option<Expr>,
+}
+
+trait Prebuild {
+    fn prebuild(self) -> Result<NodeData>;
+}
+
+fn build_stmt(select_data: NodeData) -> Statement {
```

Comment:
[Question]
Is there a reason to use just function without struct impl? Maybe we could implement this function under the struct `NodeData`.

Reply:
That is really a good point. Thanks a lot! I've just applied this